packages/fxa-auth-server/fxa-oauth-server/lib/jwt-token.js

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const hex = require('buf').to.hex;
+const AppError = require('./error');
+const JWTool = require('fxa-jwtool');
+
+const keys = require('../lib/keys');
+
+/**
+ * Create a JWT access token from `grant`
+ */
+exports.create = async function generateJWTAccessToken(db, grant) {
+  const accessToken = await db.generateAccessToken(grant);
+  const clientId = hex(grant.clientId);
+
+  // Claims list from:
+  // https://tools.ietf.org/html/draft-bertocci-oauth-access-token-jwt-00#section-2.2
+  const claims = {
+    // The IETF spec for `aud` refers to https://openid.net/specs/openid-connect-core-1_0.html#IDToken
+    // > REQUIRED. Audience(s) that this ID Token is intended for. It MUST contain the
+    // > OAuth 2.0 client_id of the Relying Party as an audience value. It MAY also contain
+    // > identifiers for other audiences. In the general case, the aud value is an array of
+    // > case sensitive strings. In the common special case when there is one audience, the
+    // > aud value MAY be a single case sensitive string.
+
+    // TODO: accept a "resource" parameter to endpoints where access tokens are returned.
```

Comment:
File a follow on for this, leave a comment about why it's just the clientId and not a URL.